edx_dl/edx_dl.py

```diff
@@ -326,7 +365,8 @@ def extract_units(url, headers):
     """
     Parses a webpage and extracts its resources e.g. video_url, sub_url, etc.
     """
-    compat_print("Processing '%s'" % url)
+    logging.info("Processing '%s'", url)
```

Comment:
On Jul 24 2015, Ismael Mejia wrote:
> > @@ -326,7 +365,8 @@ def extract_units(url, headers):
> >      """
> >      Parses a webpage and extracts its resources e.g. video_url, sub_url, etc.
> >      """
> > -    compat_print("Processing '%s'" % url)
> > +    logging.info("Processing '%s'", url)
> 
> I really think we must change this to be part of the debug information in
> a future commit.

I agree 100%.

> However I don't know how to show the progress to the user in a user
> friendly way (e.g. a progress bar or text '## / TOTAL')

I had a small, silly proof-of-concept here in one of my computers (yes, I
use more than one computer to write my code---all running Debian, though)
and I don't remember if I `git stash`ed those changes or if I simply ignored
it.

In fact, it wasn't a progress bar *per se*, but something along the lines of
"processed xxx/yyy pages."

I guess that a barebones progress bar could be stolen from `coursera-dl`'s
internal downloader. It even includes some unit tests.

OTOH, there are some full featured progress bar implementations in some
python packages on PyPI (with colors!), but, perhaps, we can use something
simpler for now.


Thanks for the comment,

-- 
Rogério Brito : rbrito@{ime.usp.br,gmail.com} : GPG key 4096R/BCFCAAAA
http://cynic.cc/blog/ : github.com/rbrito : profiles.google.com/rbrito
DebianQA: http://qa.debian.org/developer.php?login=rbrito%40ime.usp.br